case_studies/CARD/Bug-Drug Code.R

```diff
@@ -0,0 +1,148 @@
+# config.R
+url <- "https://card.mcmaster.ca/download/0/broadstreet-v3.3.0.tar.bz2"
+destfile <- "broadstreet-v3.3.0.tar.bz2"
+
+# Download the file
+download.file(url, destfile)
+
+#Extract the file
+if (!require("R.utils")) {
+  install.packages("R.utils")
+  library(R.utils)
+}
+
+
+# Extract the tar file
+untar("broadstreet-v3.3.0.tar", exdir = "CARD_data")
+
+
+# Map CARD Short Name
+# Install and Load dplyr
+if (!require("dplyr")) {
+  install.packages("dplyr")
+  library(dplyr)
+} else {
+  library(dplyr)
+}
+
+#  Read the required files
+aro_index <- read.delim("CARD_data/aro_index.tsv", sep = "\t", header = TRUE)
+antibiotics_data <- read.delim("CARD_data/shortname_antibiotics.tsv", sep = "\t", header = TRUE)
+pathogens_data <- read.delim("CARD_data/shortname_pathogens.tsv", sep = "\t", header = TRUE)
+
+
+# Extract pathogen, gene, drug, and include Protein.Accession from 'CARD.Short.Name'
+aro_index_clean <- aro_index %>%
+  mutate(
+    pathogen = sapply(strsplit(CARD.Short.Name, "_"), `[`, 1),  # Extract pathogen
```

Comment:
if possible, switch `sapply` to `purrr::map` functions
cc: @AbhirupaGhosh @epbrenner